k8s/k8s.go

```diff
@@ -88,3 +101,47 @@ func GetSecret(k8sClient *kubernetes.Clientset, secretName string) (*v1.Secret,
 func DeleteNode(k8sClient *kubernetes.Clientset, nodeName string) error {
 	return k8sClient.Nodes().Delete(nodeName, &metav1.DeleteOptions{})
 }
+
+func RunKubectlCmd(dClient *client.Client, hostname string, cmd []string, withEnv []string) error {
+
+	logrus.Debugf("[Kubectl] Using host [%s] for deployment", hostname)
+	logrus.Debugf("[Kubectl] Pulling kubectl image..")
+
+	if err := docker.PullImage(dClient, hostname, KubectlImage); err != nil {
+		return err
+	}
+	env := []string{}
+	if withEnv != nil {
+		env = append(env, withEnv...)
+	}
+	imageCfg := &container.Config{
+		Image: KubectlImage,
+		Env:   env,
+		Cmd:   cmd,
+	}
+	logrus.Debugf("[Kubectl] Creating kubectl container..")
+	resp, err := dClient.ContainerCreate(context.Background(), imageCfg, nil, nil, KubctlContainer)
+	if err != nil {
+		return fmt.Errorf("Failed to create kubectl container on host [%s]: %v", hostname, err)
+	}
+	logrus.Debugf("[Kubectl] Container %s created..", resp.ID)
+	if err := dClient.ContainerStart(context.Background(), resp.ID, types.ContainerStartOptions{}); err != nil {
+		return fmt.Errorf("Failed to start kubectl container on host [%s]: %v", hostname, err)
+	}
+	for {
+		isDeployerRunning, err := docker.IsContainerRunning(dClient, hostname, KubctlContainer)
```

Comment:
Can we explore other ways of checking if the container has done its job? Not only the container should stop, we should check the error code.

Reply:
Like subscribing to the container event. Also there should be a timeout so we don't end up waiting on things forever